### PR TITLE
Fixing calendar action in news creation

### PR DESCRIPTION
**Pull request by gnunicorn**
While working on #2089 , I noticed that the way we construct the calendar-event-inline action is ... well, wrong. Instead of the typed `CalendarReference` we were using the `LinkRef` with some weird `title` and `id` properties as strings. 

This PR corrects that with fallback support for the types we have already sent/published. See the new one following by an old one:

https://github.com/user-attachments/assets/2f287be3-c42d-4958-9c44-fd5dd303ca9a

To the reviewer:
- This refactors the code to make it easier to add more types later as well as to use proper typing
- Due to the fallback support needed, this also already contains a way to render regular links, though we don't have any construction code for it yet.
- as I did that as a split out of #2089 this is probably harder to review. You only need to care about d5050d65a3e108ae1c5a14285df19501a01ad994 and following. But we have to wait with merging until we have the other one finalised anyways.

### CODE REVIEW

File: app/lib/features/news/widgets/news_item.dart [hunk 1]
```diff
@@ -188,6 +189,13 @@ class _NewsItemState extends ConsumerState<NewsItem> {
                 L10n.of(context).eventDeletedOrFailedToLoad,
                 style: Theme.of(context).textTheme.labelLarge,
               ),
+              onTap: () async {
+                await ActerErrorDialog.show(
+                  context: context,
+                  error: e,
+                  stack: s,
+                );
+              },
```
Comment (kumarpalsinh25) [inline app/lib/features/news/widgets/news_item.dart]: 👏

File: app/lib/features/news/actions/submit_news.dart [hunk 2]
```diff
@@ -0,0 +1,175 @@
+import 'package:acter/common/providers/sdk_provider.dart';
```
Comment (kumarpalsinh25) [inline app/lib/features/news/actions/submit_news.dart:1]: Assuming that this is just separate of function code into separate action file, not changes in code logic.

File: app/lib/features/news/model/news_references_model.dart [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+}
+
+String _toCamelCase(String s) {
+  final words = s
+      .split('-')
+      .map((w) =>
+          '${w.substring(0, 1).toUpperCase()}${w.substring(1).toLowerCase()}',)
+      .toList();
+  words[0] = words[0].toLowerCase();
+  return words.join();
```
Comment (kumarpalsinh25) [inline app/lib/features/news/model/news_references_model.dart:17]: Not quite understand this code.

File: app/lib/features/news/widgets/news_item.dart [hunk 4]
```diff
@@ -169,44 +172,101 @@ class _NewsItemState extends ConsumerState<NewsItem> {
...
+    return renderActionButton(referenceDetails);
+  }
+
+  Widget renderActionButton(RefDetails referenceDetails) {
+    final evtType = NewsReferencesType.fromStr(referenceDetails.typeStr());
+
+    return switch (evtType) {
+      NewsReferencesType.calendarEvent => renderCalendarEventAction(
+          targetEventId: referenceDetails.targetIdStr() ?? '',),
+      NewsReferencesType.link => renderLinkActionButtion(referenceDetails),
```
Comment (kumarpalsinh25) [inline app/lib/features/news/widgets/news_item.dart:185]: Not quite understand the use-case of `Link`, is that was for backend support?

File: app/lib/features/news/widgets/news_item.dart [hunk 5]
```diff
@@ -169,44 +172,101 @@ class _NewsItemState extends ConsumerState<NewsItem> {
...
+      _ => renderNotSupportedAction()
+    };
+  }
+
+  Widget renderLinkActionButtion(RefDetails referenceDetails) {
```
Comment (kumarpalsinh25) [inline app/lib/features/news/widgets/news_item.dart:190]: Same as above, not quite understand this use-case.

**Review by kumarpalsinh25 — APPROVED** [review]
Review below commits which are specific to this PR.
- https://github.com/acterglobal/a3/pull/2096/commits/d5050d65a3e108ae1c5a14285df19501a01ad994
- https://github.com/acterglobal/a3/pull/2096/commits/69b97ce6361fd504c7ea1bd6ed5e5e919ef1c844

In general changes looks good but there were few things that either I can't remember or understand well. Happy to proceed with merge after answering the queries.

File: app/lib/features/news/widgets/news_item.dart (hunk 4 above)
Comment (gnunicorn) [reply app/lib/features/news/widgets/news_item.dart:185]: Yes, that was what we did before - but it was wrong

File: app/lib/features/news/widgets/news_item.dart (hunk 5 above)
Comment (gnunicorn) [reply app/lib/features/news/widgets/news_item.dart:190]: To support messages we have already sent (and can't alter anymore) in the older wrong format

File: app/lib/features/news/widgets/news_item.dart (hunk 4 above)
Comment (gnunicorn) [reply app/lib/features/news/widgets/news_item.dart:185]: In general Link is for any external URL to anything else.

File: app/lib/features/news/model/news_references_model.dart (hunk 3 above)
Comment (gnunicorn) [reply app/lib/features/news/model/news_references_model.dart:17]: it rewrites strings from `things-named-like-this` to `thingsNamedLikeThis`

File: app/lib/features/news/actions/submit_news.dart (hunk 2 above)
Comment (gnunicorn) [reply app/lib/features/news/actions/submit_news.dart:1]: yes.

**Pull request merged**